platforms/nuttx/src/px4/stm/stm32f4/px4io_serial/px4io_serial.cpp

```diff
@@ -294,49 +289,55 @@ ArchPX4IOSerial::_bus_exchange(IOPacket *_packet)
 
 	/* wait for the transaction to complete - 64 bytes @ 1.5Mbps ~426µs */
 	int ret;
+	irqstate_t irqs = enter_critical_section();
 
 	for (;;) {
 		ret = sem_timedwait(&_completion_semaphore, &abstime);
 
 		if (ret == OK) {
 			/* check for DMA errors */
 			if (_rx_dma_status & DMA_STATUS_TEIF) {
-				// stream transfer error, ensure all DMA is also stopped before exiting early
-				_abort_dma();
+				/* stream transfer error, ensure all DMA is also stopped before exiting early */
```

Comment:
Comment is not correct anymore. one of 3 things  has happed. 1. a DMA Stream error. 2. Serial parity, framing or over run error. 3. packet is malformed.  In all cased DMA is stopped by either HW or the ISR error service path.

Reply:
Done